### ISSUE TITLE
Reset doesn't work for Background field.

### ISSUE THREAD

**Issue by vlthemes** [themeum/kirki#1524]
Hi Ari.

I found nothing in the documentation for this field. 
How do I set the default image?

```php
Kirki::add_field( 'ramsay_customize', array(
    'type' => 'background',
    'settings' => 'footer_background',
    'label' => esc_html__( 'Footer Background', 'ramsay' ),
    'description' => esc_html__( 'Select background for footer', 'ramsay' ),
    'section' => 'section_footer_styling',
    'priority' => $priority++,
    'default' => array(
        'background-color' => '#252525',
        'background-image' => $theme_path_images . 'footer-bg.png',
        'background-size' => 'cover'
    ),
    'output' => array(
        array(
            'element' => '.vl-footer-outer',
        ),
    )
) );
```

**Comment by carasmo** [comment]
Use the [stylesheet directory uri](https://codex.wordpress.org/Function_Reference/get_stylesheet_directory_uri) or a constant you define.

`'background-image'		 => get_stylesheet_directory_uri() . '/assets/images/yourdefault.jpg',`

**Comment by vlthemes** [comment]
This works for me here and I dont think that the problem in this

```php
$theme_path_images = RAMSAY_THEME_DIRECTORY . 'assets/img/';
Kirki::add_field( 'ramsay_customize', array(
    'type' => 'image',
    'settings' => 'footer_logo',
    'label' => esc_html__( 'Footer logo', 'ramsay' ),
    'description' => esc_html__( 'Choose a logo image to display for footer', 'ramsay' ),
    'section' => 'section_footer_logo',
    'default' => $theme_path_images . 'logo-white.png',
    'transport' => 'auto',
    'priority' => $priority++,
) );
```

**Comment by carasmo** [comment]
This is what I use. You can look in the code base to figure out most stuff if there is no documentation. Also, the _s fork stuff to learn with.

	My_Theme_Class::add_field( 'my_prefix', array( 
	'type'         => 'background',
	'settings'     => 'body_bg',
	'label'        => __( 'Body Background Image', 'text-domain' ),
	'section'      => 'theme_prefix_background_section',
	'transport'    => 'auto',
		'default'  => array(
			'background-image'       => CHILD_URL . '/assets/images/body-bg.jpg',
			'background-color'       => '#f3ebe2',
			'background-repeat'      => 'repeat-all',
			'background-size'        => 'auto',
			'background-attachment'  => 'fixed',
			'background-position'    => 'left top',
		),
        'output'  => array( 
            array( 
                'element' => 'body',
             ),
         ),
	) );

**Comment by vlthemes** [comment]
Thank you so much. It works.

I did not know that the rest of the fields are required.
```php
'background-repeat'      => 'repeat-all',
'background-size'        => 'auto',
'background-attachment'  => 'fixed',
'background-position'    => 'left top',
```

**Comment by carasmo** [comment]
Also, transport auto will do the postMessage js work.

**Comment by vlthemes** [comment]
So, when I click Reset button nothing happens. Dont reset background-position, size etc.

**Comment by aristath** [comment]
@vlthemes using stable version from wp.org or the develop branch?
If you haven't tried the develop branch can you please test that?

**Comment by vlthemes** [comment]
Recent dev version.

**Comment by vlthemes** [comment]
I used this code.

```php
$priority = 0;
Kirki::add_field( 'ramsay_customize', array(
	 'type' => 'background',
	'settings' => 'footer_background',
	'label' => esc_html__( 'Footer Background', 'ramsay' ),
	'description' => esc_html__( 'Select background for footer', 'ramsay' ),
	'section' => 'section_footer_styling',
	'priority' => $priority++,
	'default' => array(
		 'background-image' => $theme_path_images . 'footer-bg.png',
		'background-color' => '#252525',
		'background-repeat' => 'no-repeat',
		'background-size' => 'cover',
		'background-attachment' => 'scroll',
		'background-position' => 'top left' 
	),
	'output' => array(
		 array(
			 'element' => '.vl-footer-outer' 
		) 
	) 
) );
```

**Comment by aristath** [comment]
Does the `$theme_path_images` var contain the trailing slash?
When using urls in vars like you do it's always advisable to use tailingslashit:
`trailingslashit( $theme_path_images ) . 'footer-bg.png'`
Do a `var_dump( $theme_path_images )` to see what you've got there exactly.
If you open the browser console when viewing that control in the customizer, do you see any errors there?

**Comment by vlthemes** [comment]
There are no mistakes and the path to the image is correct. Just does not work reset button on the field of background.

**Comment by aristath** [comment]
oh, so the field works fine and the only problem is reset, correct?

**Comment by vlthemes** [comment]
yeah. I use this code:
```php
$priority = 0;
Kirki::add_field( 'ramsay_customize', array(
	'type' => 'background',
	'settings' => 'footer_background',
	'label' => esc_html__( 'Footer Background', 'ramsay' ),
	'description' => esc_html__( 'Select background for footer', 'ramsay' ),
	'section' => 'section_footer_styling',
	'priority' => $priority++,
	'default' => array(
		'background-image' => trailingslashit( $theme_path_images ) . 'footer-bg.png',
		'background-color' => '#252525',
		'background-repeat' => 'no-repeat',
		'background-size' => 'cover',
		'background-attachment' => 'scroll',
		'background-position' => 'top left' 
		),
	'output' => array(
		array(
			'element' => '.vl-footer-outer' 
			) 
		) 
	) );
```

**Comment by vlthemes** [comment]
And I have some error in console log Did not parse stylesheet at 'http://localhost:8888/wordpress/assets/css/kirki-styles.css' because non CSS MIME types are not allowed in strict mode.

**Comment by aristath** [comment]
OK, that last comment seems to be shedding some light on what's going on here...
`http://localhost:8888/wordpress/assets/css/kirki-styles.css` is not a valid URL! It should be `http://localhost:8888/wordpress/wp-content/plugins/kirki/assets/css/kirki-styles.css`

Is kirki installed as a plugin, or embedded in a theme?
Which version are you using? You mentioned in a previous comment "recent dev version"... how recent? Can you please try the latest one from https://github.com/aristath/kirki/archive/develop.zip ?

**Comment by vlthemes** [comment]
Embedded.
Recent dev.

**Comment by aristath** [comment]
Can you please be more specific? What do you mean "recent"?
Is it last month's? Last week's? or did you just pull the develop branch just now?

**Comment by vlthemes** [comment]
Version 3.0.10-beta.1

**Comment by aristath** [comment]
Please pull the latest develop branch.
I don't change the version number on every commit I make, so the version can be the same for months while the code is different.

**Comment by vlthemes** [comment]
I just updated it and nothing has changed.
http://take.ms/UohPB


Is there a problem with that?
http://take.ms/rwp6u

**Comment by aristath** [comment]
1. It's strongly advised not to embed the plugin in your theme. Instead you can advise or require your users to install it as a plugin There are many ways to do that and be compatible with everything, have fallbacks in case Kirki is not installed etc.
2. That function should not be necessary at all. Try this: In your functions.php file add this code:
```php
add_action( 'wp_footer', function(
	var_dump( 'Kirki URL: ' . Kirki::$url );
	var_dump( 'Kirki Path: ' . Kirki::$path );
) );
```
Visit your site, see the footer, and take note what it says there.
Then delete the custom filter you have that changes the url, and see if that changes the values in your footer.

If that doesn't work, try installing kirki as a plugin and see if that works (leave the custom filter deleted).

**Comment by vlthemes** [comment]
Problem still exists
http://take.ms/fKSeA

It is not convenient for me to use kirki as a plugin.

**Comment by aristath** [comment]
The issue here seems to be that somehow the URL is not properly detected, and that's causing the issues.

> It is not convenient for me to use kirki as a plugin.

When asking for assistance or reporting an issue, it is advisable to provide as many information as possible. You are using kirki not as it was intended to be used (as a plugin) but instead embed it in a theme and also use a custom filter to alter the URL internally. That's all ok, but these are things that should have been mentioned from the beginning. I would have been able to pinpoint the issue - and resolve it - a lot sooner if I had known these things 9 days ago when you initially opened this ticket. It would have saved us both a lot of time.

There is no way for me to debug something properly when I don't have all the details and only have bits and pieces of information.

If it's not convenient for you to use kirki, then just don't use it.

**Comment by vlthemes** [comment]
I apologize for the inconvenience.
In 3.0.9 there was no error, I do not know, it's yours.

**Comment by aristath** [comment]
> In 3.0.9 there was no error, I do not know, it's yours.

I'm not saying there is no error, I never said that. Yes, there is an error and yes, it's mine.
But without some basic details like the fact that you're on MAMP and I only saw that from a screenshot of your path, plus that it's embedded in a theme made detecting the error impossible for me.
I was only able to replicate this now when I tried using MAMP and embedding the plugin in a theme. On Vagrant or an actual linux host this doesn't happen. When using kirki as a plugin this doesn't happen either. It's a combination that was impossible for me to imagine.
